pkg/server/server_test.go

```diff
@@ -171,6 +224,46 @@ var _ = Describe("ApplicationsServer", func() {
 				vals := log.WithValuesArgsForCall(0)
 				list := formatLogVals(vals)
 
+				expectedStatus := strconv.Itoa(res.StatusCode)
+				Expect(list).To(ConsistOf("uri", path, "status", expectedStatus))
+			})
+			It("Authorize fails generating jwt token", func() {
+
+				fakeJWTToken := &authfakes.FakeJWTClient{}
+				fakeJWTToken.GenerateJWTStub = func(s2 string, duration time.Duration, name gitproviders.GitProviderName, s22 string) (string, error) {
+					return "", fmt.Errorf("some error")
+				}
+
+				appsSrv = NewApplicationsServer(&ApplicationConfig{App: app.New(nil, nil, nil, kubeClient, nil), JwtClient: fakeJWTToken})
+				mux = runtime.NewServeMux(middleware.WithGrpcErrorLogging(log))
+				httpHandler = middleware.WithLogging(log, mux)
+				err = pb.RegisterApplicationsHandlerServer(context.Background(), mux, appsSrv)
+				Expect(err).NotTo(HaveOccurred())
+
+				ts := httptest.NewServer(httpHandler)
+				defer ts.Close()
+
+				// A valid URL for our server
+				path := "/v1/authenticate/github"
+				url := ts.URL + path
+
+				res, err := http.Post(url, "application/json", strings.NewReader("{\"accessToken\":\"sometoken\"}"))
+				Expect(err).NotTo(HaveOccurred())
+				Expect(res.StatusCode).To(Equal(http.StatusInternalServerError))
+
+				response := make(map[string]interface{})
```

Comment:
You could probably use [`MatchJSON`](https://onsi.github.io/gomega/#matchjsonjson-interface) here